server/endpoints/Review.ts

```diff
@@ -0,0 +1,60 @@
+import { body } from "express-validator";
+import { getCrossListOR } from "common/CourseCard";
+import { Endpoint } from "../endpoints";
+import { Reviews, Classes } from "../dbDefs";
+
+interface CourseId {
+  courseId: string;
+}
+
+/**
+ * Get a course with this course_id from the Classes collection in the local
+ * database.
+ */
+export const getCourseById: Endpoint<CourseId> = {
+  guard: [body("courseId").notEmpty().isAscii()],
+  callback: async (courseId: CourseId) => {
+    try {
+      // check: make sure course id is valid and non-malicious
+      const regex = new RegExp(/^(?=.*[A-Z0-9])/i);
+      if (regex.test(courseId.courseId)) {
+        return await Classes.findOne({ _id: courseId.courseId }).exec();
+      }
+      return null;
+    } catch (error) {
+      // eslint-disable-next-line no-console
+      console.log("Error: at 'getCourseById' method");
+      // eslint-disable-next-line no-console
+      console.log(error);
+      return null;
```

Comment:
I would return some sort of error message for these nulls. See how I did it in the example in Search.ts: e.g ```return { error: "Malformed Query" };```

This goes for several other places where null is returned. Let me know if you disagree though.

Reply:
Agree here, error descriptions are generally helpful